File: python/src/Structural/Facade/FacadeCuppaMaker.py
from FacadeTeaCup import FacadeTeaCup
from FacadeTeaBag import FacadeTeaBag
from FacadeWater import FacadeWater

class FacadeCuppaMaker:

	teaBagIsSteeped  = None

	def makeACuppa(self) :
		cup = FacadeTeaCup()
		teaBag = FacadeTeaBag()
		water = FacadeWater()
		cup.addFacadeTeaBag(teaBag)
		water.boilFacadeWater()
		cup.addFacadeWater(water)
		cup.steepTeaBag()
		return cup

File: ruby/src/Structural/Facade/FacadeCuppaMaker.rb
#require "FacadeTeaCup"
#require "FacadeTeaBag"
#require "FacadeWater"

class FacadeCuppaMaker

	def initialize()
		@teaBagIsSteeped = false
	end

	def makeACuppa() 		
		cup = FacadeTeaCup.new()
		teaBag = FacadeTeaBag.new()
		water = FacadeWater.new()
		cup.addFacadeTeaBag(teaBag)
		water.boilFacadeWater()
		cup.addFacadeWater(water)
		cup.steepTeaBag()
		return cup
	end

end

